### PR TITLE
Updating SwiftSyntax to Swift 5.2

### LINKED ISSUE #6: Xcode 11.4 will not build.
SwiftSyntax parser library isn't compatible
Command PhaseScriptExecution failed with a nonzero exit code

### PR DESCRIPTION
**Pull request by nerdsupremacist**
fixes #6 

Merge when SwiftSyntax gets a `0.50200.0` tag

### CODE REVIEW

File: Sources/Graphaello/Processing/Pipeline/Component/2. Parsing/SubParsers/SubParser+functionCall+Argument.swift [hunk 1]
```diff
@@ -13,11 +13,11 @@ extension SubParser {
     
     static func functionCall(parser: @escaping () -> SubParser<ExprSyntaxProtocol, Argument.QueryArgument>) -> SubParser<FunctionCallExprSyntax, Argument> {
         return .init { expression in
-            guard let calledMember = expression.calledExpression.asProtocol(ExprSyntaxProtocol.self) as? MemberAccessExprSyntax else {
+            guard let calledMember = expression.calledExpression.withoutErasure() as? MemberAccessExprSyntax else {
                 throw ParseError.cannotInstantiateObjectFromExpression(expression, type: Argument.self)
             }
 
-            let argument = Array(expression.argumentList).single()?.expression
+            let argument = Array(expression.argumentList).single()?.expression.withoutErasure()
```
Comment (ahoppen) [inline Sources/Graphaello/Processing/Pipeline/Component/2. Parsing/SubParsers/SubParser+functionCall+Argument.swift]: You should not need to call `withoutErasure()` here. Without looking too much into the details here, I would suggest that the `parse(from:)` function on `SubParser` takes a `Syntax` or `ExprSyntax` (I don't know the exact context) as an argument. Inside `parse(from:)`, you can then switch on the exact type of the node by
- using `as(SyntaxEnum.self)` (or `as(Syntax.self).as(SyntaxEnum.self)` if `parse(from:)` takes an `ExprSyntax`
- using `as(SyntaxProtocol.self)` or `as(ExprSyntaxProtocol.self)`

Comment (nerdsupremacist) [reply Sources/Graphaello/Processing/Pipeline/Component/2. Parsing/SubParsers/SubParser+functionCall+Argument.swift]: I love it!

**Comment by nerdsupremacist** [comment]
@ahoppen thanks for the feedback, I really appreciate it...

And I think you'd like this project ;)

File: Sources/Graphaello/Extensions/ExprSyntax+erasure.swift [hunk 2]
```diff
@@ -4,8 +4,8 @@ import SwiftSyntax
 
 extension ExprSyntax {
 
-    func withoutErasure() -> ExprSyntaxProtocol {
-        return asProtocol(ExprSyntaxProtocol.self)
+    func switchOver() -> SyntaxEnum {
```
Comment (ahoppen) [inline Sources/Graphaello/Extensions/ExprSyntax+erasure.swift]: Some background on why `as(SyntaxEnum.self)` is only defined on `Syntax` and not on `ExprSyntax`: The `SyntaxEnum` contains every syntax type, while `ExprSyntax` already has the set restricted, thus we decided that it was better to make the "upcast" explicit by requiring to write `as(Syntax.self).as(SyntaxEnum.self)`.

File: Sources/Graphaello/Extensions/ExprSyntax+erasure.swift [hunk 3]
```diff
@@ -4,8 +4,8 @@ import SwiftSyntax
...
+        return _syntaxNode.as(SyntaxEnum.self)
```
Comment (ahoppen) [inline Sources/Graphaello/Extensions/ExprSyntax+erasure.swift]: `_syntaxNode` is an underscored member and should not be accessed unless there is evidence that it improves performance in performance-critical code. The preferred spelling is `Syntax(self)`

File: Sources/Graphaello/Extensions/ExprSyntax+erasure.swift (hunk 2 above)
Comment (nerdsupremacist) [reply Sources/Graphaello/Extensions/ExprSyntax+erasure.swift]: Very good point

**Pull request merged**